src/virtual_dom/update_el.rs

```diff
@@ -179,3 +128,82 @@ where
         self.for_each(|item| item.update(el));
     }
 }
+
+// ----- Containers ------
+
+impl<Ms, T> UpdateEl<El<Ms>> for Option<T>
+where
+    T: UpdateEl<El<Ms>>,
+{
+    fn update(self, el: &mut El<Ms>) {
+        if let Some(val) = self {
+            val.update(el);
+        }
+    }
+}
+
+impl<Ms, T, E> UpdateEl<El<Ms>> for Result<T, E>
+where
+    T: UpdateEl<El<Ms>>,
+    E: UpdateEl<El<Ms>>,
+{
+    fn update(self, el: &mut El<Ms>) {
+        match self {
+            Ok(val) => val.update(el),
+            Err(err) => err.update(el),
+        }
```

Comment:
I suggest to change it to:
```rust
if let Ok(val) = self {
    val.update(el);
}
```
- So it's prepared for future `UpdateEl` implementation for `IntoInterator`.
- Users are not limited by `UpdateEl` restriction for `err`.
- My intuition tells me that `Err` value should be ignored like `Options`'s `None`.

Reply:
I don't agree with this, the whole point on implementing `UpdateEl` to `Result` was to handle `Err` too.

if we didn't handle `Err` users might pass unhandled `Result` mistakenly and we simply have discard that `Err`.

Reply:
> if we didn't handle Err users might pass unhandled Result mistakenly and we simply have discard that Err.

Yes, I agree on 100%. 
But accidentally output a (long and cryptic) error to HTML which breaks layout and confuses user it even worse. It's pretty easy to do that - `String` and many other entities implements `UpdateEl`.
I begin to think that implementing `UpdateEl` for `Result` is really bad idea.

Reply:
valid argument, I can delete the implementation for `Result`.
`Result` can be easily converted to `Option` (with `result.ok()`).